anki/template/template.py

```diff
@@ -200,20 +205,27 @@ def render_unescaped(self, tag_name=None, context=None):
 
     def clozeText(self, txt, ord, type):
         reg = clozeReg
-        if not re.search(reg%ord, txt):
+        if not re.search(reg % ord, txt):
             return ""
+
         def repl(m):
             # replace chosen cloze with type
+            noSound = removeClozeSound(m.group(3))
```

Comment:
That is exactly what it means.

That was pretty ugly to start with.  How about this.

``` def repl(m):
            # replace chosen cloze with type
            hint = m.group(3)
            answer = m.group(1)
            hintWithNoSound = removeClozeSound(hint)
            if type == "q":
                if hintWithNoSound:
                    return "<span class=cloze>[%s]</span>" % hint
                elif hint:
                    return "<span class=cloze>[...]%s</span>" % hint
                else:
                    return "<span class=cloze>[...]</span>"
            else:
                return "<span class=cloze>%s</span>" % answer
```